processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -330,25 +329,67 @@ public static Long convertObjectToLong(@Nullable Object valObj, boolean reportPa
     } else if (valObj instanceof String) {
       Long ret = DimensionHandlerUtils.getExactLongFromDecimalString((String) valObj);
       if (reportParseExceptions && ret == null) {
-        throw new ParseException((String) valObj, "could not convert value [%s] to long", valObj);
+        final String message = (objectKey != null) ? StringUtils.nonStrictFormat(
+            "could not convert value [%s] to long for dimension [%s]",
+            valObj,
+            objectKey
+        ) : StringUtils.nonStrictFormat(
+            "could not convert value [%s] to long",
+            valObj
+        );
+        throw new ParseException((String) valObj, message);
       }
       return ret;
     } else if (valObj instanceof List) {
-      throw new ParseException(
-          valObj.getClass().toString(),
+      final String message = (objectKey != null) ? StringUtils.nonStrictFormat(
+          "Could not ingest value [%s] as long for dimension [%s]. A long column cannot have multiple values in the same row.",
+          valObj,
+          objectKey
+      ) : StringUtils.nonStrictFormat(
           "Could not ingest value %s as long. A long column cannot have multiple values in the same row.",
           valObj
       );
-    } else {
       throw new ParseException(
+          valObj.getClass().toString(),
+          message
+      );
+    } else {
+      final String message = (objectKey != null) ? StringUtils.nonStrictFormat(
+          "Could not convert value [%s] to long for dimension [%s]. Invalid type: [%s]",
+          valObj,
+          objectKey,
+          valObj.getClass()
+      ) : StringUtils.nonStrictFormat(
           valObj.getClass().toString(),
           "Could not convert value [%s] to long. Invalid type: [%s]",
           valObj,
           valObj.getClass()
       );
+      throw new ParseException(
+          valObj.getClass().toString(),
+          message
+      );
     }
   }
 
+  @Nullable
+  public static Long convertObjectToLong(@Nullable Object valObj)
+  {
+    return convertObjectToLong(valObj, false);
+  }
+
+  @Nullable
+  public static Long convertObjectToLong(@Nullable Object valObj, boolean reportParseExceptions)
+  {
+    return convertObjectToLong(valObj, reportParseExceptions, null);
+  }
+
+  @Nullable
+  public static Long convertObjectToLong(Object value, String fieldName)
```

Comment:
Looks like the first arg can be nullable similar to the other double and float variants (also suggest renaming it to `valObj` for consistency):
```suggestion
  public static Long convertObjectToLong(@Nullable Object valObj, String fieldName)
```